src/.clang-format

```diff
@@ -1,50 +1,96 @@
+AllowAllConstructorInitializersOnNextLine: true
 AllowAllParametersOfDeclarationOnNextLine: false
-AllowShortBlocksOnASingleLine: false
-AllowShortFunctionsOnASingleLine: All
+AllowShortBlocksOnASingleLine: Empty
+AllowShortCaseLabelsOnASingleLine: true
+AllowShortFunctionsOnASingleLine: Inline
 AllowShortIfStatementsOnASingleLine: false
+AllowShortLambdasOnASingleLine: All
 AllowShortLoopsOnASingleLine: false
 AlwaysBreakBeforeMultilineStrings: false
 AlwaysBreakTemplateDeclarations: true
+BinPackArguments: false
 BinPackParameters: false
-BreakBeforeBinaryOperators: false
+BreakBeforeBinaryOperators: NonAssignment
 BreakBeforeBraces: Linux
-BreakBeforeTernaryOperators: false
-BreakConstructorInitializersBeforeComma: false
-ColumnLimit:     0
-CommentPragmas:  '^ IWYU pragma:'
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializers: BeforeColon
+ColumnLimit: 100 # match rustfmt, if we don’t define this, then we don’t get Wadler
+CommentPragmas: '^ IWYU pragma:'
 ConstructorInitializerAllOnOneLineOrOnePerLine: false
 ConstructorInitializerIndentWidth: 4
 ContinuationIndentWidth: 4
 Cpp11BracedListStyle: true
 DerivePointerAlignment: false
-DisableFormat:   false
-IndentCaseLabels: false
+DisableFormat: false
+IncludeBlocks: Regroup
+IncludeCategories:
+    - Regex:           '<[^/.]+>'
+      Priority:        3
+    - Regex:           '<.*>'
+      Priority:        2
+    - Regex:           '.*'
+      Priority:        1
+IndentCaseLabels: true
 IndentFunctionDeclarationAfterType: false
-IndentWidth:     4
+IndentWidth: 4
+InsertBraces: true
 KeepEmptyLinesAtTheStartOfBlocks: false
-Language:        Cpp
+Language: Cpp
 MaxEmptyLinesToKeep: 2
 NamespaceIndentation: None
 ObjCSpaceAfterProperty: false
 ObjCSpaceBeforeProtocolList: false
-PenaltyBreakBeforeFirstCallParameter: 1
-PenaltyBreakComment: 300
-PenaltyBreakFirstLessLess: 120
-PenaltyBreakString: 1000
-PenaltyExcessCharacter: 1000000
-PenaltyReturnTypeOnItsOwnLine: 200
+PackConstructorInitializers: NextLine
+# PenaltyBreakAssignment: 0
+# PenaltyBreakBeforeFirstCallParameter: 1
+# PenaltyBreakComment: 300
+# PenaltyBreakFirstLessLess: 120
+PenaltyBreakOpenParenthesis: 1
+# PenaltyBreakString: 1000
+# PenaltyExcessCharacter: 1000000
+PenaltyReturnTypeOnItsOwnLine: 0
 PointerAlignment: Left
 SpaceBeforeAssignmentOperators: true
+SpaceBeforeCpp11BracedList: true
+SpaceBeforeCtorInitializerColon: true
 SpaceBeforeParens: ControlStatements
 SpaceInEmptyParentheses: false
+SpaceAfterTemplateKeyword: false
 SpacesBeforeTrailingComments: 1
-SpacesInAngles:  false
-SpacesInContainerLiterals: true
+SpacesInAngles: false
+SpacesInContainerLiterals: false
 SpacesInCStyleCastParentheses: false
 SpacesInParentheses: false
-Standard:        Cpp11
-TabWidth:        8
-UseTab:          Never
+Standard: c++17
+TabWidth: 8
+UseTab: Never
+# This is a bit of a hack to get clang-format to let us do what we want with examine, even though
+# it’s not a macro.
+WhitespaceSensitiveMacros: ['examine']
```

Comment:
This can be deleted if we use [AlignAfterOpenBracket: BlockIndent](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#alignafteropenbracket), I think. Not sure how disruptive that would be.